### PR TITLE
Bug 1462773: Add Processes parser

### PR DESCRIPTION
**Pull request by sunahsuh**
Adding a simple parser for [`Processes.yaml`](https://hg.mozilla.org/releases/mozilla-release/raw-file/tip/toolkit/components/telemetry/Processes.yaml) here so we can stop trying to keep lists up to date in both t-b-v and telemetry-streaming. When we switch over to using the probe scraper service as described in the bug, we should be able to swap out the backend while keeping the (very simple) interface consistent.

Don't hesitate to go to town on this -- I'd love any and all feedback on improving style, etc

### CODE REVIEW

File: src/main/scala/com/mozilla/telemetry/pings/main/Processes.scala [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+class ProcessesClass {
+  protected case class Process(name: String, geckoEnum: Option[String], description: Option[String])
+  protected val getURL: (String, String) => scala.io.Source = Source.fromURL
+
+  protected lazy val parsedProcesses: Map[String, Seq[Process]] = {
+    val uris = Map("release" -> "https://hg.mozilla.org/releases/mozilla-release/raw-file/tip/toolkit/components/telemetry/Processes.yaml",
+      "beta" -> "https://hg.mozilla.org/releases/mozilla-beta/raw-file/tip/toolkit/components/telemetry/Processes.yaml",
+      "nightly" -> "https://hg.mozilla.org/mozilla-central/raw-file/tip/toolkit/components/telemetry/Processes.yaml")
+    val sources = uris.map { case (key, value) => key -> getURL(value, "UTF8") }
```
Comment (akkomar) [inline src/main/scala/com/mozilla/telemetry/pings/main/Processes.scala]: Assuming these processes don't change often, shouldn't we fetch these mappings once and cache in this repo?
We can create an object or script for downloading these to `resources` and commit here. If keeping them in sync is essential, we can create an sbt task for checking if cached files are the same as those in hg repo and run it as part of CI build.

I think if we keep this implementation:
* we might have problems with local testing t-b-v and t-s without internet access
* production jobs will fail if `hg.mozilla.org` will be inaccessible from their hosts

Comment (sunahsuh) [reply src/main/scala/com/mozilla/telemetry/pings/main/Processes.scala]: They don't, and I do strongly prefer not having an internet dependency (we have a bug for that https://bugzilla.mozilla.org/show_bug.cgi?id=1387091) but 1. this is patterned after the [Histograms/Scalars code in telemetry-batch-view](https://github.com/mozilla/telemetry-batch-view/blob/master/src/main/scala/com/mozilla/telemetry/metrics/Histogram.scala#L111-L113) so these are already issues that we have and 2. that's all being refactored out (and migrated here to moztelemetry, iirc) to use the probe-scraper, at some point (which doesn't really mitigate either of your concerns, but it's definitely worth bringing up your thoughts on in that context.)

Basically, adding a periodic script to cache this file right now would be a fair amount of overhead that wouldn't really mitigate your concerns because the same issues are present elsewhere, and adding new code to do this is probably outside the scope of this PR, which is aimed toward not missing new process types in the several places that use them.

File: src/test/scala/com/mozilla/telemetry/pings/main/ProcessesTest.scala [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+package com.mozilla.telemetry.pings.main
+
+import org.scalatest.{FlatSpec, Matchers}
+import scala.io.Source
+
+class ProcessesTest extends FlatSpec with Matchers {
+  val fixture = {
```
Comment (akkomar) [inline src/test/scala/com/mozilla/telemetry/pings/main/ProcessesTest.scala]: nit: do we need a `fixture` since there's only one test here?
